deploy/objects/Deployer.js

```diff
@@ -1,62 +1,62 @@
-  },
-
-  async createProxy(projectController, contractClazz, contractName, distribution, version) {
-    const controllerManager = new ProjectControllerManager(controller, owner);
-    return await controllerManager.createProxy(contractClazz, contractName, distribution, version);
-  },
-
-  async createProxyAndCall(controller, owner, contractKlazz, contractName, distribution, version, initArgTypes, initArgs) {
-    const controllerManager = new ProjectControllerManager(controller, owner);
-    return await controllerManager.createProxyAndCall(contractKlazz, contractName, distribution, version, initArgTypes, initArgs);
-  },
-
-  async zepCore(owner, newVersionCost, developerFraction, distribution = 'ZeppelinOS') {
-    const controller = await this.projectController(owner, distribution)
-    const registry = Registry.at(await controller.registry())
-    await this._registerZepCoreDependencies(registry);
-    const controllerManager = new ProjectControllerManager(controller, owner);
-    const kernelStakes = await controllerManager.createProxyAndCall(KernelStakes, 'KernelStakes', distribution, '0', ['address'], [owner]);
-    const kernelRegistry = await controllerManager.createProxyAndCall(KernelRegistry, 'KernelRegistry', distribution, '0', ['address'], [owner]);
-    const zepToken = await controllerManager.createProxyAndCall(ZepToken, 'ZepToken', distribution, '0', ['address'], [owner]);
-    const zepCore = await controllerManager.createProxyAndCall(ZepCore, 'ZepCore', distribution, '0',
-      ['uint256', 'uint256', 'address', 'address', 'address'],
-      [newVersionCost, developerFraction, zepToken.address, kernelRegistry.address, kernelStakes.address]
+class Deployer {
+  constructor(owner) {
+    this.owner = owner;
+  }
+
+  async initAppManager(initialVersion) {
+    const factory = await UpgradeabilityProxyFactory.new({ from: this.owner });
+    this.package = await Package.new({ from: this.owner });
+    this.contractDirectory = await ContractDirectory.new(0, { from: this.owner });
+    await this.package.addVersion(initialVersion, this.contractDirectory.address, { from: this.owner });
+    this.appManager = await AppManager.new(this.package.address, initialVersion, factory.address, { from: this.owner });
+  }
+
+  async atAppManager(address) {
+    // TODO: Implement me to set this.(package, contractDirectory, appManager, appManagerWrapper)
+  }
+
+  async newVersion(versionName) {
+    this.contractDirectory = await ContractDirectory.new(0, { from: this.owner });
+    await this.package.addVersion(versionName, this.contractDirectory.address, { from: this.owner });
+    this.appManager.setVersion(versionName, { from: this.owner });
+  }
+
+  async deployAndRegister(contractClass, contractName) {
+    const implementation = await contractClass.new({ from: this.owner });
+    await this.contractDirectory.setImplementation(contractName, implementation.address, { from: this.owner });
```

Comment:
we should use `AppManager#getProvider` here, you should not know which is the current one